geomstats/geometry/pullback_metric.py

```diff
@@ -125,34 +174,94 @@ def inner_product_derivative_matrix(self, base_point=None):
         inner_prod_deriv_mat : array-like, shape=[..., dim, dim, dim]
             Inner-product derivative matrix.
         """
-        hessian_aij = []
-        jacobian_ai = []
-        embedding_dim = self.embedding_metric.dim
-        for a in range(embedding_dim):
+        initial_ndim = base_point.ndim
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
+        inner_prod_deriv_mats = []
+        for point in base_point:
+            jacobian_ai = self.jacobian_immersion(point)
+            if self.dim == 1 and jacobian_ai.ndim > 2:
```

Comment:
These ifs are not obvious. I think we could gain by adding a comment here (I also have the feeling they can somehow be avoided by increasing consistency in the code output in other regions of the code base).